tasks/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -110,24 +110,37 @@ spec:
         echo "${pyxisKey}" > /tmp/key
 
         COMPONENTS=$(jq '.components | length' "${SNAPSHOT_SPEC_FILE}")
+        JSON_OUTPUT='{}'
         for (( i=0; i < $COMPONENTS; i++ )); do
             CONTAINER_IMAGE=$(jq -r ".components[${i}].containerImage" "${SNAPSHOT_SPEC_FILE}")
+            JSON_OUTPUT=$(jq ".components[$i] += { \"containerImage\": \"${CONTAINER_IMAGE}\", \"pyxisImages\": []}" \
+                              <<< "$JSON_OUTPUT")
             REPOSITORY=$(jq -r ".components[${i}].repository" "${SNAPSHOT_SPEC_FILE}")
             REPOSITORY=${REPOSITORY%:*} # strip tag just in case - it should not be there
             DIGEST="${CONTAINER_IMAGE##*@}"
             PULLSPEC="docker://${REPOSITORY}@${DIGEST}"
             MEDIA_TYPE=$(skopeo inspect --raw "$PULLSPEC" | jq -r .mediaType)
-            skopeo inspect --no-tags "$PULLSPEC" > /tmp/skopeo-inspect.json
+            ARCHS=($(get-image-architectures "${PULLSPEC#*://}"))
+            for (( a=0; a < ${#ARCHS[@]}; a++)); do
+                ARCH=${ARCHS[$a]}
+                skopeo inspect --no-tags --override-os linux --override-arch "$ARCH" "$PULLSPEC" \
+                    | tee /tmp/skopeo-inspect.json
+                PYXIS_CERT_PATH=/tmp/crt PYXIS_KEY_PATH=/tmp/key create_container_image \
+                  --pyxis-url $PYXIS_URL \
+                  --certified $(params.certified) \
+                  --tags "$TAGS" \
+                  --is-latest $(params.isLatest) \
+                  --verbose \
+                  --skopeo-result "/tmp/skopeo-inspect.json" \
+                  --media-type "$MEDIA_TYPE" \
+                  --rh-push $(params.rhPush) | tee "/tmp/output"
+                JSON_OUTPUT=$(jq \
+                    ".components[$i].pyxisImages[$a] += {
+                      \"arch\": \"${ARCH}\",
+                      \"imageId\": \"$(awk '/The image id is/{print $NF}' /tmp/output)\",
+                      \"digest\": \"${DIGEST}\"}" <<< "$JSON_OUTPUT")
 
-            PYXIS_CERT_PATH=/tmp/crt PYXIS_KEY_PATH=/tmp/key create_container_image \
-              --pyxis-url $PYXIS_URL \
-              --certified $(params.certified) \
-              --tags "$TAGS" \
-              --is-latest $(params.isLatest) \
-              --verbose \
-              --skopeo-result /tmp/skopeo-inspect.json \
-              --media-type "$MEDIA_TYPE" \
-              --rh-push $(params.rhPush) | tee /tmp/output
-
-            grep 'The image id is' /tmp/output | awk '{print $NF}' >> $(results.containerImageIDs.path)
+                grep 'The image id is' /tmp/output | awk '{print $NF}' >> $(results.containerImageIDs.path)
+            done
         done
+        jq . <<< "$JSON_OUTPUT"  |tee $(workspaces.data.path)/pyxis.json
```

Comment:
`$JSON_OUTPUT` was already produced by `jq`, so no need to process it with it again. Instead, you can just do:
`tee $(workspaces.data.path)/pyxis.json <<< "$JSON_OUTPUT"` or `echo "$JSON_OUTPUT" | tee $(workspaces.data.path)/pyxis.json`